extension/src/prototypes/view/ui-containers/DatasetFolerItem.tsx

```diff
@@ -0,0 +1,67 @@
+import { IconButton } from "@mui/material";
+import { FC, useCallback, useState, type MouseEvent } from "react";
+
+import { DatasetFragmentFragment } from "../../../shared/graphql/types/catalog";
+import { DatasetTreeItem, InfoIcon } from "../../ui-components";
+
+import { DatasetDialog } from "./DatasetDialog";
+import { DatasetFolderList } from "./DatasetFolderList";
+
+export type FolderItem = {
+  label: string;
+  subFolderId: string;
+  datasets: DatasetFragmentFragment[];
+  folderDataset?: DatasetFragmentFragment;
+};
+
+type DatasetFolderItemProps = {
+  folderItem: FolderItem;
+  level: number;
+};
+
+export const DatasetFolderItem: FC<DatasetFolderItemProps> = ({ folderItem, level }) => {
+  const [infoOpen, setInfoOpen] = useState(false);
+  const handleInfo = useCallback((event: MouseEvent) => {
```

Comment:
The folder might not have the description. So I guess we should prevent to open the dialog if the folder doesn't have the description.